src/test/java/org/openhab/tools/analysis/checkstyle/test/PomXmlCheckTest.java

```diff
@@ -98,18 +99,31 @@ public void testMasterPom() throws Exception {
         verify(createChecker(config), testFileAbsolutePath, ArrayUtils.EMPTY_STRING_ARRAY);
     }
 
+    @Test
+    public void testDifferentVersion() throws Exception {
+        verifyFile("different_version_directory", false, 0,
+                "The manifest version did not match the requirements " + MANIFEST_REGEX);
+    }
+
     @Override
     protected DefaultConfiguration createCheckerConfig(Configuration config) {
         DefaultConfiguration defaultConfiguration = new DefaultConfiguration("root");
         defaultConfiguration.addChild(config);
         return defaultConfiguration;
     }
 
-    private void verifyPomXmlFile(String testDirectoryName, int expectedLine, String expectedMessage) throws Exception {
+    private void verifyFile(String testDirectoryName, boolean isPomXmlFile, int expectedLine, String expectedMessage)

```

Comment:
Adding a boolean argument makes these tests harder to read and understand.

Could you just add a separate method `verifyManifestFile` ?